src/librustdoc/clean/inline.rs

```diff
@@ -162,12 +162,11 @@ crate fn load_attrs<'hir>(cx: &DocContext<'hir>, did: DefId) -> Attrs<'hir> {
 /// These names are used later on by HTML rendering to generate things like
 /// source links back to the original item.
 crate fn record_extern_fqn(cx: &mut DocContext<'_>, did: DefId, kind: ItemType) {
-    let crate_name = cx.tcx.crate_name(did.krate).to_string();
+    let crate_name = cx.tcx.crate_name(did.krate);
 
     let relative = cx.tcx.def_path(did).data.into_iter().filter_map(|elem| {
         // extern blocks have an empty name
-        let s = elem.data.to_string();
-        if !s.is_empty() { Some(s) } else { None }
+        elem.data.get_opt_name().and_then(|s| if !s.is_empty() { Some(s) } else { None })
```

Comment:
I don't think this `is_empty` check is necessary since it will be handled by `get_opt_name`.

Reply:
cc #92032: I'm not sure what we should be doing here.

Reply:
When both PRs are merged this will turn into just `elem.data.get_opt_name()`.